### PR TITLE
NIFI-4082 - Added EL on GetMongo properties

### PR DESCRIPTION
**Pull request by pvillard31**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by jfrazee** [comment]
@pvillard31 I think we can add EL to collection without having to do anything fancy since getCollection() in AbstractMongoProcessor re-uses the mongoClient already.

That said, is it worth also adding EL support for URI and DB in this PR? The simple case will mean it'll get eval'd against the variable registry, which is definitely useful for migrating from env to env.

There's another case though in being able to eval against a FlowFile for PutMongo so you can route to different DBs if you're doing some application-level sharding. But that would mean we'd have to manage a pool/map of connections since the mongoClient currently is set with @OnScheduled. Makes sense if you think that's for another PR though.

**Comment by pvillard31** [comment]
@jfrazee - I just pushed a new commit and added support for expression language with FFs evaluation on DB and Collection name, and without FFs evaluation on URI. Regarding this very subject, what are your thoughts about NIFI-4149 and how this could solve this kind of need?

**Comment by jfrazee** [comment]
@pvillard31 I left some comments on [NIFI-4149](https://issues.apache.org/jira/browse/NIFI-4149). I think it'd be a step in the right direction.

File: nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java [hunk 1]
```diff
@@ -59,14 +59,21 @@
...
+
+            if (context.isExpressionLanguageSupported(subject) && context.isExpressionLanguagePresent(value)) {
+                return builder.valid(true).explanation("Contains Expression Language").build();
+            }
+
             String reason = null;
             try {
                 Document.parse(value);
             } catch (final RuntimeException e) {
                 reason = e.getClass().getName();
```
Comment (jfrazee) [inline nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java]: Seems like this should be `e.getMessage()` or `get.getClass().getName() + ": " + e.getMessage()` if you want the underlying exception class.

**Comment by jfrazee** [comment]
@pvillard31 I think this needs more updates to the tests. Since you've added URI, collection, and DB, we should probably add that to the Get test. And, I think there are similar changes that make sense for the Put and Abstract processor.

**Comment by pvillard31** [comment]
@jfrazee just pushed a commit to address your last comment: changed the exception message for the custom document validator, and added unit tests for the expression language evaluation on URI, database name and collection name properties.

**Comment by nicholasmhughes** [comment]
@pvillard31 @jfrazee I'm very interested in this functionality. The Jira for this issue doesn't have a fix version identified. Do you think it will make it into 1.4.0 (or the next formal release)?

**Comment by mattyb149** [comment]
+1 LGTM, ran builds/tests and tested GetMongo and PutMongo with EL for all fields, talked to Joey and he's cool with me merging too. Thanks for the improvements! Merging to master

**Pull request closed without merging**